### PR TITLE
Update operator-lint to latest

### PR DESCRIPTION
**Pull request by amolkahat**
This will fix the pre-commit failures.

openshift/release: [36102](https://github.com/openshift/release/pull/36012)

Signed-off-by: Amol Kahat <amolkahat@gmail.com>

### CODE REVIEW

File: config/crd/bases/neutron.openstack.org_neutronapis.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.10.0
+    controller-gen.kubebuilder.io/version: v0.9.2
```
Comment (sandeepyadav93) [inline config/crd/bases/neutron.openstack.org_neutronapis.yaml:6]: Just curious, what's the reason behind this?

Comment (amolkahat) [reply config/crd/bases/neutron.openstack.org_neutronapis.yaml:6]: This got updated after pre-commit run.

Comment (karelyatin) [reply config/crd/bases/neutron.openstack.org_neutronapis.yaml:6]: can you try again after dropping rm bin/controller-gen from local neutron-operator clone. With that these files will not change, v0.10.0 is the current version as per CONTROLLER_TOOLS_VERSION ?= v0.10.0 in Makefile

File: api/bases/neutron.openstack.org_neutronapis.yaml [hunk 2]
```diff
@@ -113,7 +113,7 @@ spec:
                   database: NeutronDatabasePassword
                   service: NeutronPassword
                 description: PasswordSelectors - Selectors to identify the DB and
-                  ServiceUser and NoveService User password from the Secret
+                  ServiceUser password from the Secret
```
Comment (karelyatin) [inline api/bases/neutron.openstack.org_neutronapis.yaml:116]: same here

File: Makefile [hunk 3]
```diff
@@ -302,5 +302,5 @@ gowork: ## Generate go.work file to support our multi module repository
 
 .PHONY: operator-lint
 operator-lint: gowork ## Runs operator-lint
-	GOBIN=$(LOCALBIN) go install github.com/gibizer/operator-lint@v0.1.0
+	GOBIN=$(LOCALBIN) go install github.com/gibizer/operator-lint@latest
```
Comment (karelyatin) [inline Makefile:305]: from past discussion i recall intentions were to use pinned version of operator-lint so projects can move to more recent version whenver it suits to project requirement. or the decision changed or i missed something?

can you also share the Error which this change is fixing to understand better.

**Review by karelyatin — CHANGES_REQUESTED** [review]

File: config/crd/bases/neutron.openstack.org_neutronapis.yaml (hunk 1 above)
Comment (amolkahat) [reply config/crd/bases/neutron.openstack.org_neutronapis.yaml:6]: Well this works.

File: Makefile (hunk 3 above)
Comment (amolkahat) [reply Makefile:305]: Well fixing some erros with pre-commit.. This is not relevant patch to the errors. Closing.

**Comment by amolkahat** [comment]
Closing. not relevant.

**Pull request closed without merging**